src/commands/Minion/blocklist.ts

```diff
@@ -0,0 +1,149 @@
+			usage: '<taskname:...string>',
+			usageDelim: ' ',
+			oneAtTime: true,
+			cooldown: 1,
+			altProtection: true
+		});
+	}
+
+	async run(msg: KlasaMessage, [taskname]: [string]) {
+		await msg.author.settings.sync(true);
+		if (msg.author.minionIsBusy) {
+			return msg.send(msg.author.minionStatus);
+		}
+		const userBlockList = msg.author.blockList;
+		const allTasks = nieveTasks.concat(turaelTasks);
+		// Block if their current task matches the block request
+		const task = allTasks.filter(task => stringMatches(taskname, task.name));
+		if (taskname.toLowerCase() === Monsters.get(msg.author.slayerTaskID)?.name.toLowerCase()) {
+			if (userBlockList.length >= 5) {
+				throw `You already have a full block list`;
+			}
+			if (msg.author.slayerPoints < 100) {
+				throw `You need 100 slayer points to block that task and you only have ${msg.author.slayerPoints}`;
+			}
+			msg.send(`Are you sure you'd like to block ${taskname}? Say \`confirm\` to continue.`);
+			try {
+				await msg.channel.awaitMessages(
+					_msg =>
+						_msg.author.id === msg.author.id &&
+						_msg.content.toLowerCase() === 'confirm',
+					options
+				);
+			} catch (err) {
+				throw `Cancelling block list addition of ${taskname}.`;
+			}
+			await msg.author.removeSlayerPoints(100);
+			await msg.author.settings.update(UserSettings.Slayer.SlayerTaskQuantity, 0);
+			await msg.author.settings.update(UserSettings.Slayer.HasSlayerTask, false);
+			await msg.author.settings.update(UserSettings.Slayer.SlayerTaskID, 0);
+			await msg.author.settings.update(UserSettings.Slayer.CurrentSlayerMaster, 0);
+			await msg.author.settings.update(UserSettings.Slayer.BlockList, task[0].ID, {
+				arrayAction: 'add'
+			});
+			return msg.send(
+				`The task **${taskname}** has been **added** to your block list. You have ${msg.author.slayerPoints} Slayer Points left. Your current task of ${taskname} has also been cancelled.`
+			);
+		}
+		if (taskname === 'show') {
+			let str = 'Your current block list: ';
+			for (let i = 0; i < userBlockList.length; i++) {
+				const blocked = Monsters.get(userBlockList[i]);
+				const blockedName = blocked?.name;
+				str += `${blockedName}, `;
+			}
+			str = str.replace(/,\s*$/, '');
+			if (userBlockList.length === 0) {
+				throw `You don't have any blocked tasks yet`;
+			}
+			throw str;
+		}
+
+		// Block list removal
+		if (msg.flagArgs.unblockthistask) {
+			if (!userBlockList.includes(task[0].ID)) {
+				throw `That task isn't on your block list.`;
+			}
+			msg.send(
+				`Are you sure you'd like to unblock ${taskname}? Say \`confirm\` to continue.`
+			);
+			try {
+				await msg.channel.awaitMessages(
+					_msg =>
+						_msg.author.id === msg.author.id &&
+						_msg.content.toLowerCase() === 'confirm',
+					options
+				);
+			} catch (err) {
+				throw `Cancelling block list removal of ${taskname}.`;
+			}
+			await msg.author.settings.update(UserSettings.Slayer.BlockList, task[0].ID, {
+				arrayAction: 'remove'
```

Comment:
Use the enum for the array actions from Klasa, in all occurrences (Array actions)